ui/app/src/components/DependencyItem.tsx

```diff
@@ -0,0 +1,206 @@
+  root: {
+    justifyContent: 'center',
+    color: '#7e9dbb',
+  }
+})(CircularProgress);
+
+const BlueCheckbox = withStyles({
+  root: {
+    color: '#7e9dbb',
+    padding: '4px 6px',
+    '&$checked': {
+      color: '#7e9dbb',
+    },
+  },
+  checked: {},
+})(Checkbox);
+
+const BlueExpandMoreIcon = withStyles({
+  root: {
+    color: '#7e9dbb'
+  }
+})(ExpandMoreIcon);
+
+function DependecyItem(props: DependecyItemProps) {
+
+  const [deps, setDeps] = useState<DepsObject>();
+
+  useEffect(
+    () => {
+      if(props.isOpen && deps === undefined){
+        get(`/studio/api/2/dependency/dependencies?siteId=${props.siteId}&paths=${props.item.uri}`)
+          .subscribe(
+            (response: any) => {
+              setDeps({
+                hard: response.response.items.hardDependencies,
+                soft: response.response.items.softDependencies
+              });
+            },
+            () => {
+              setDeps({
+                hard: [],
+                soft: []
+              });
+            }
+          );
+      }
+    },
+    [props.isOpen]
+  );
+
+  return (
+    <div>
+      <ExpansionPanel square expanded={props.isOpen} onChange={(e, nextVisible) => props.onToggle(nextVisible)}>
+        <ExpansionPanelSummary expandIcon={<BlueExpandMoreIcon />} aria-controls="panel1d-content" id="">
+          <Typography>
+            <BlueCheckbox
+              checked={!!props.checked[props.item.uri]}
+              onClick={(e:any) => {
+                e.stopPropagation();
+                e.preventDefault();
+                props.onSelection(props.item.uri, !props.checked[props.item.uri])
+              }}
+              onChange={(e) => {}}
+              value={props.item.uri}
+              color="primary"
+            />
+            
+            <strong>
+              <span>{props.item.internalName} •</span>
+              <span>&nbsp;{props.item.uri}</span>
+            </strong>
+          </Typography>
+        </ExpansionPanelSummary>
+        <ExpansionPanelDetails>
+            {
+              deps == null ? (
+                <div className="centerCircularProgress">
+                  <CenterCircularProgress />
+                </div>
+              ) : (
+                <div>
+                  <p className="hardDependenciesTitle" ><strong>Hard Dependencies</strong></p>
+                  <ul>
+                    {
+                      deps && deps.hard
+                      ? ( 
+                        deps.hard.map((uri: string) => (
+                          <li key={uri}>{uri}</li>
+                        ))
+                      ) 
+                      : (
+                        <div></div>
+                      )
+                    }
+                  </ul>
+                    <p>
+                      <strong>Soft Dependencies</strong> • <span>Submission Optional</span> 
+                      <button className="navButtons" onClick={selectAllSoft}>Select All</button>
```

Comment:
Dasherized classnames. This class is probably not even used so you should remove it completely if that's the case.

Reply:
file is removed